File: packages/camera/camera_android/android/src/test/java/io/flutter/plugins/camera/features/videostabilization/VideoStabilizationModeTest.java
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.camera.features.videostabilization;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

import org.junit.Test;

import io.flutter.plugins.camera.features.autofocus.FocusMode;

public class VideoStabilizationModeTest {

  @Test
  public void getValueForString_returnsCorrectValues() {
    assertEquals(
        "Returns VideoStabilizationMode.off for 'off'", VideoStabilizationMode.getValueForString("off"), VideoStabilizationMode.off);
    assertEquals(
            "Returns VideoStabilizationMode.on for 'on'", VideoStabilizationMode.getValueForString("on"), VideoStabilizationMode.on);
    assertEquals(
            "Returns VideoStabilizationMode.previewStabilization for 'previewStabilization'", VideoStabilizationMode.getValueForString("previewStabilization"), VideoStabilizationMode.previewStabilization);
  }

  @Test
  public void getValueForString_returnsNullForNonexistantValue() {
      assertNull("Returns null for 'nonexistant'", VideoStabilizationMode.getValueForString("nonexistant"));
  }

  @Test
  public void toString_returnsCorrectValue() {
    assertEquals("Returns 'off' for VideoStabilizationMode.off", VideoStabilizationMode.off.toString(), "off");
    assertEquals("Returns 'on' for VideoStabilizationMode.on", VideoStabilizationMode.on.toString(), "on");
    assertEquals("Returns 'previewStabilization' for VideoStabilizationMode.previewStabilization", VideoStabilizationMode.previewStabilization.toString(), "previewStabilization");
  }
}

File: packages/camera/camera_platform_interface/test/types/video_stabilization_mode_test.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:camera_platform_interface/src/types/video_stabilization_mode.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('VideoStabilizationMode should contain 3 options', () {
    const List<VideoStabilizationMode> values = VideoStabilizationMode.values;

    expect(values.length, 3);
  });

  test('VideoStabilizationMode enum should have items in correct index', () {
    const List<VideoStabilizationMode> values = VideoStabilizationMode.values;

    expect(values[0], VideoStabilizationMode.off);
    expect(values[1], VideoStabilizationMode.on);
    expect(values[2], VideoStabilizationMode.previewStabilization);
  });

  test('serializeVideoStabilizationMode() should serialize correctly', () {
    expect(serializeVideoStabilizationMode(VideoStabilizationMode.off), 'off');
    expect(serializeVideoStabilizationMode(VideoStabilizationMode.on), 'on');
    expect(
        serializeVideoStabilizationMode(
            VideoStabilizationMode.previewStabilization),
        'preview_stabilization');
  });

  test('deserializeVideoStabilizationMode() should deserialize correctly', () {
    expect(
        deserializeVideoStabilizationMode('off'), VideoStabilizationMode.off);
    expect(deserializeVideoStabilizationMode('on'), VideoStabilizationMode.on);
    expect(deserializeVideoStabilizationMode('preview_stabilization'),
        VideoStabilizationMode.previewStabilization);
  });
}
